### ISSUE TITLE
Is it normal for the learning rate to reach the peak_value and not decrease, but instead slowly rise?

### ISSUE THREAD

**Issue by joytianya** [young-geng/EasyLM#22]
https://api.wandb.ai/links/matrix-zxw/qmcnboxa
Is it normal for the learning rate to reach the peak_value 1e-4 and not decrease, but instead slowly rise?

python -m EasyLM.models.llama.llama_train \
--optimizer.type=adamw \
--optimizer.adamw_optimizer.lr=1e-4 \
...

**Comment by young-geng** [comment]
Have you configured the lr_decay_steps for the adamw optimizer?

**Comment by joytianya** [comment]
I use the default value, warmup step 2000, delay step 500000

**Comment by joytianya** [comment]
config.lr_warmup_steps = 2000
config.lr_decay_steps = 500000

**Comment by joytianya** [comment]
https://github.com/young-geng/EasyLM/blob/main/EasyLM/optimizers.py#L137
See the details here, are the default values configured here?

**Comment by ja5087** [comment]
Make sure to configure `optimizer.adamw_optimizer.end_lr`. By default it is `1e-3`

**Comment by joytianya** [comment]
https://wandb.ai/matrix-zxw/llama-7b--all/reports/learning_rate-23-04-09-08-26-27---Vmlldzo0MDEwNTA4
@ja5087 Thank you for your answer. 
I have modified the corresponding configuration, and this time it reached the peak value but remained unchanged.
Is there any configuration that is incorrect?

LEARN_RATE=2e-5
WARMUP_STEPS=3000
END_LEARN_RATE=0.0
WEIGHT_DECAY=0.0
    
    --optimizer.adamw_optimizer.lr=${LEARN_RATE} \
    --optimizer.adamw_optimizer.lr_warmup_steps=$WARMUP_STEPS \
    --optimizer.adamw_optimizer.end_lr=${END_LEARN_RATE} \
    --optimizer.adamw_optimizer.weight_decay=${WEIGHT_DECAY} \

**Comment by joytianya** [comment]
https://wandb.ai/matrix-zxw/llama-7b--all/reports/learning_rate-23-04-09-08-26-27---Vmlldzo0MDEwNTA4
@ja5087 Thank you for your answer. 
I have modified the corresponding configuration, and this time it reached the peak value but remained unchanged.
Is there any configuration that is incorrect?

LEARN_RATE=2e-5
WARMUP_STEPS=3000
END_LEARN_RATE=0.0
WEIGHT_DECAY=0.0
    
    --optimizer.adamw_optimizer.lr=${LEARN_RATE} \
    --optimizer.adamw_optimizer.lr_warmup_steps=$WARMUP_STEPS \
    --optimizer.adamw_optimizer.end_lr=${END_LEARN_RATE} \
    --optimizer.adamw_optimizer.weight_decay=${WEIGHT_DECAY} \

**Comment by young-geng** [comment]
You probably also need to change the `lr_decay_steps`. That controls when the learning rate will decay to the end value.

**Comment by joytianya** [comment]
ok, thank you~
